lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,195 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "Runtime.h"
+
+#include <time.h>
+#include <sys/time.h>
+
+#if HAVE_MACH_ABSOLUTE_TIME
+#include <mach/mach_time.h>
+static mach_timebase_info_data_t s_TimebaseInfo;
+#endif
+
+#include "Platform/Common/DateTime.h"
+
+namespace PlatformAgnostic {
+namespace DateTime {
+
+    #define updatePeriod 1000
+
+    // Buffer size is compatible to Windows _TIME_ZONE_INFORMATION
+    static WCHAR zoneName[32] = {0};
+    static size_t zoneNameLength = 0;
+    static uint32 lastTimeZoneUpdateTickCount = 0;
+
+    static void UpdateTimeZoneInfo() noexcept
+    {
+        uint32 tickCount = GetTickCount();
+        if (tickCount - lastTimeZoneUpdateTickCount > updatePeriod)
+        {
+            time_t gtime = time(NULL);
+            struct tm ltime;
+
+            tzset();
+            localtime_r(&gtime, &ltime);
+
+            zoneNameLength = strlen(ltime.tm_zone);
+
+        #if defined(WCHAR_IS_CHAR16_T)
```

Comment:
What scenario do we have to support non-char16 scenarios? Seems fine to always have a char16 implementation (and assert for that at compile time)